examples/plugins/provision.py

```diff
@@ -84,6 +84,7 @@ def go(self):
 
         self._guest = GuestExample(data, name=self.name, parent=self.step)
         self._guest.start()
+        self._guest.setup()
```

Comment:
Every plugin already calls `start()`. Can `start()` call `setup()` directly? That would give us just one method to remember to call in a plugin, instead of two.

`setup()` follows `start()`, with the exception of the `local` and `connect` plugins, and I can imagine these two could add their custom "start nothing and call `start()`, like the parent would" implementations.

Reply:
@happz , I thought of having this operation done in `start` directly or indirectly with another method. However, I read the description of `start` and hesitated. It appears to me that adding this functionality, which gets inside the machine itself to do things, increases the semantic of the `start` function, perhaps in a surprising way for a developer.

`start` a `guest` seems to be that the end goal is to have a guest turned on and access given, but the internal modification in order to prepare doesn't come to mind. `setup` a `guest`, on the other hand, can imply an internal modification of the VM. Naming and semantics definition is one of the hardest things in computer science, lol.

Thoughts?

Reply:
> There are 2 hard problems in computer science: cache invalidation, naming things, and off-by-1 errors.

Anyway. Good point. I still think not giving developers a chance to forget such an important step in the process is worth the effort :) I'll think about it, maybe we could extend the semantics of `Guest.start()`, maybe rename it, who knows.

Reply:
I'm looking more into this one.

The "start" in artemis and mrack means "create". In test cloud means the provisioning.  The "setup" step here is really a "post" start step common to all of them. So we get then a "prestart" that does the create, or provisioning, and a "poststart" that does what I have called "setup". Whoever comes later with a new implementation of "start" will need to remember to also run the "poststart" after the machine is created.

Another place where I could put this is in the `PreparePlugin`, which has a `go` method which is always called in the subclasses via `super`.